modules/imgproc/src/color_rgb.simd.hpp

```diff
@@ -138,8 +138,13 @@ struct RGB2RGB
                 v_load_deinterleave(src, a, b, c);
                 d = v_set<_Tp>::set(alphav);
             }
-            if(bi == 2)
+            if(bi == 2) {
+                #if CV_SIMD_SCALABLE
+                auto t = a; a = c; c = t; // swap(a, c);
+                #else
                 swap(a, c);
```

Comment:
do we really need the swap branch?

Reply:
@asmorkalov , apparently `std::swap` does not support RVV data types.
```
/opt/riscv64/gcc-10.2.0/riscv64-unknown-linux-gnu/lib/gcc/riscv64-unknown-linux-gnu/10.2.0/../../../../riscv64-unknown-linux-gnu/include/c++/10.2.0/type_traits:960:7: error: static assertion failed due to requirement 'std::__is_complete_or_unbounded(std::__type_identity<__rvv_int32m1_t>{})': template argument must be a complete class or an unbounded array
      static_assert(std::__is_complete_or_unbounded(__type_identity<_Tp>{}),
      ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
`swap` exists in the original algorithm implementation, the modification here is that swap cannot used for the RVV type (as @mshabunin says above), so we use `auto t = a; a = c; c = t;` to replace it.

I haven't read the algorithm itself in its entirety, so I can't say why there are `swap` in the original algorithm. But I tried to remove all the three `swap`, the result of the is wrong, therefore I think `swap` is necessary.

```
[  FAILED  ] 1 test, listed below:
[  FAILED  ] Imgproc_ColorRGB.accuracy
```